### PR TITLE
Implement PUT requests for VA Profile when processing opt in/out preference updates.

### PR DESCRIPTION
**Pull request by kalbfled**
#679

This branch is based on #635, which currently has a separate PR.

I encountered pre-commit errors for reasons I still don't understand, and I updated the pre-commit repo versions to try to address them.

### CODE REVIEW

File: lambda_functions/va_profile/va_profile_opt_in_out_lambda.py [hunk 1]
```diff
@@ -6,9 +6,15 @@
 import os
 import psycopg2
 import sys
+from http.client import ConnectionError, HTTPSConnection
+from json import dumps
+
 
 OPT_IN_OUT_QUERY = """SELECT va_profile_opt_in_out(%s, %s, %s, %s, %s);"""
 NOTIFICATION_API_DB_URI = os.getenv("notification_api_db_uri")
+VA_PROFILE_DOMAIN = "https://api.va.gov"
```
Comment (kalbfled) [inline lambda_functions/va_profile/va_profile_opt_in_out_lambda.py:15]: This isn't in the VA Profile sync document, and I'm not sure that it is correct.

Comment (jakehova) [reply lambda_functions/va_profile/va_profile_opt_in_out_lambda.py:15]: I agree.  Was there a domain specified in the docs?  I know we used a route and defined the method.

File: lambda_functions/va_profile/va_profile_opt_in_out_lambda.py [hunk 2]
```diff
@@ -71,7 +77,31 @@ def va_profile_opt_in_out_lambda_handler(event: dict, context) -> dict:
...
+            if record.get("txAuditId", '') != event["txAuditId"]:
+                raise KeyError
+
+            put_record = {
+                "vaProfileId": record["vaProfileId"],
+                "communicationChannelId": record["communicationChannelId"],
+                "communicationItemId": record["communicationItemId"],
+            }
+        except KeyError:
+            problem_detected = True
```
Comment (jakehova) [inline lambda_functions/va_profile/va_profile_opt_in_out_lambda.py:101]: Does this get logged anywhere?

**Review by jakehova — APPROVED** [review]
The KeyError exception not getting logged is my only open question.  I'm guessing that there is some python trick that may not require it there but figured I would point it out.

File: lambda_functions/va_profile/va_profile_opt_in_out_lambda.py (hunk 1 above)
Comment (kalbfled) [reply lambda_functions/va_profile/va_profile_opt_in_out_lambda.py:15]: It's not in the doc.  I e-mailed Josh L. and got the answer I need, but now I need to update the infra code with some new variables.

**Review by jakehova — APPROVED** [review]

**Review by EvanParish — APPROVED** [review]

**Comment by kalbfled** [comment]
PR 703 supercedes this PR.

**Pull request closed without merging**